### ISSUE TITLE
Error "simple-bar-index.jsx: Something went wrong..." with custom yabaiPath

### ISSUE THREAD

**Issue by TSOT7** [Jean-Tinland/simple-bar#401]
After modifying the yabaiPath variable in simple-bar-index.jsx to "/opt/homebrew/bin/yabai", I encounter an error message at the top of my screen that reads "simple-bar-index.jsx: Something went wrong...". This error is occuring no matter what I do. I am on a m2 air.

<img width="1470" alt="image" src="https://github.com/Jean-Tinland/simple-bar/assets/34929161/16755b41-f35b-4c01-8518-d13290efd082">

**Comment by Jean-Tinland** [comment]
Did you use the correct yabai path? It should be the output of `which yabai`.

**Comment by TSOT7** [comment]
Yes it is correct, I ran which yabai and it matches the path I have in index.jsx.

**Comment by adrien-barret** [comment]
same error (m3 sonoma)

**Comment by ppiwo** [comment]
I had to update it here as well
https://github.com/Jean-Tinland/simple-bar/blob/master/lib/settings.js#L529

**Comment by Jean-Tinland** [comment]
This is weird as the line you referred in your message is the default setting which should be overridden by this setting:
<img width="464" alt="image" src="https://github.com/Jean-Tinland/simple-bar/assets/43068795/0ea65228-0846-4b00-a6bf-5cb35da5920b">

I'll let you close this issue if everything is working fine now :)

**Comment by AnirudhG07** [comment]
I am not sure if it is already taken into consideration, by cant the path be taken from the output of `which yabai` or `which aerospace`. I had to change the path for aerospace, it worked fine. But having it present automatically seems more convenient for everyone!

**Comment by Jean-Tinland** [comment]
Thanks @AnirudhG07! I indeed tried to go this way at first, but in the `init-(yabai|aerospace).sh` script, `which yabai` or `which aerospace` commands aren't printing anything.

Maybe I'm missing something, but manually specifying the used window manager binary path is supposed to always work.

**Comment by AnirudhG07** [comment]
Hi, I just tried simply replace `$1` with `$(which aerospace)` and reloaded simple-bar and I didnt face any issues. Can you share where all I need to try these changes to test the above myself.

**Comment by Jean-Tinland** [comment]
It may work for the majority of users, but not all. I found this issue (#63) opened back in 2020 in which we tried to make it work with `$(which yabai)`.

I can take a look at it again if this is really a pain point for everyone but I'd rather work on other bugs or new features :)

**Comment by coderzawad** [comment]
@Jean-Tinland Why dont you just hard code the path for yabaiPath Var and make it like $(which yabai) or make a script to find the path for yabai?

**Comment by Jean-Tinland** [comment]
@AnirudhG07 & @coderzawad

It may not be ideal, but I'd like to truly test this, so I pushed some changes on the master branch: the custom path setting for yabai and AeroSpace has both been removed.

Both the init scripts are now using $(which yabai) or $(which aerospace) and it seems to work well. Same for utility scripts. You'll need to fully restart Übersicht after pulling simple-bar.

Please let me know if you see any issue with this update.

**Comment by AnirudhG07** [comment]
No it doesnt work. It breaks from [6deccae0](https://github.com/Jean-Tinland/simple-bar/commit/6deccae077f9196fc58bb938890429aab180be32).

**Comment by Jean-Tinland** [comment]
Oh, okay, it seemed to work well for me.

Can you please check the Übersicht console > "Show debug console" > "Foreground" window :
<img width="240" alt="image" src="https://github.com/user-attachments/assets/dca1b98f-b78b-4b7c-96a2-063940d09534" />

Is there any specific error there?

**Comment by AnirudhG07** [comment]
![40660](https://github.com/user-attachments/assets/ff4f332b-b716-436c-bfdd-3647fbfcae1b)

I am getting this, not sure how to interpret this

**Comment by Jean-Tinland** [comment]
Thanks!

It seems that you have enable simple-bar-server connection but it isn't running.
<img width="285" alt="image" src="https://github.com/user-attachments/assets/88d4cf10-723e-4a5a-b3b0-aaaacbed4d79" />
Try disabling this and look the console again?

**Comment by AnirudhG07** [comment]
Not working again.

![91011](https://github.com/user-attachments/assets/88b95cf7-e240-4799-87f5-befa9c9143bc)

**Comment by Jean-Tinland** [comment]
Okay, thanks! I don't know why this isn't working so I came up with an hybrid solution: yabai and AeroSpace path can still be setup in settings but the default values are `$(which yabai)` and `$(which aerospace)`.

This way it should work out of the box for most people but anyone having problems can manually set the path in settings.

Let me know if everything is working again.

**Comment by AnirudhG07** [comment]
Not working :<. I guess drop those commits.

**Comment by Jean-Tinland** [comment]
Can you please manually check out to the latest commit that works? I'd like to be sure it's working before reverting all changes :)

**Comment by AnirudhG07** [comment]
@Jean-Tinland [this](https://github.com/Jean-Tinland/simple-bar/commit/14ba6a1e8547b444c5667de9cb586b6db1b8b370) one.

**Comment by Jean-Tinland** [comment]
It seems that the init script fails and returns "aerospaceError" that is catched by the index.jsx in order to display the error message.

Can you try to run this command in your terminal and see if it succeed?

```shell
aerospace list-monitors --json --format "%{monitor-id} %{monitor-appkit-nsscreen-screens-id}" 2> /dev/null
```

This is the failing command in `init-aerospace.sh` in your case.

**Comment by Jean-Tinland** [comment]
Are you using the latest AeroSpace version? Knowing the error is coming from this, I think the latest changes I made should have worked.

What is the output of `aerospace -v` in your terminal?

Mine is:
```text
aerospace CLI client version: 0.16.2-Beta 21336ad382539b35fdc94b4fbd55408e10b101f8
AeroSpace.app server version: 0.16.2-Beta 21336ad382539b35fdc94b4fbd55408e10b101f8
```

**Comment by coderzawad** [comment]
@Jean-Tinland You should make a discord server for issues and other stuff its way easier to maintain there

**Comment by Jean-Tinland** [comment]
You are not wrong, but I'd rather use the GitHub discussions if possible; that way, I only have one support ecosystem to check!

Anyway, I'm closing this issue as this error is generally showing when a user has a wrong window manager path defined in settings. I also added a comment in `index.jsx` where people were trying to fix the error by changing a fallback value without effect. This comment should ideally redirect the user to the settings module.
